argoverse/evaluation/eval_detection.py

```diff
@@ -0,0 +1,402 @@
+    affinity_fn_type: AffFnType = AffFnType.CENTER
+    n_rec_samples: int = 101
+    tp_thresh: float = 2.0  # Meters
+    detection_classes: List[str] = list(OBJ_CLASS_MAPPING_DICT.keys())
+    detection_metric: FilterMetric = FilterMetric.EUCLIDEAN
+    max_detection_range: float = 100.0  # Meters
+    save_figs: bool = False
+    tp_normalization_terms: np.ndarray = np.array([tp_thresh, 1.0, MAX_YAW_ERROR])
+
+
+class DetectionEvaluator(NamedTuple):
+    """Instantiates a DetectionEvaluator object for evaluation.
+
+    Args:
+        detection_fpath: The path to the folder which contains the detections.
+        ground_truth_fpath: The path to the folder which contains all the logs.
+        figures_fpath: The path to the folder which will contain the output figures.
+        detection_cfg: The detection configuration settings.
+    """
+
+    detection_fpath: Path
+    ground_truth_fpath: Path
+    figures_fpath: Path
+    detection_cfg: DetectionCfg = DetectionCfg()
+
+    def evaluate(self) -> pd.DataFrame:
+        """Evaluate detection output and return metrics. The multiprocessing
+        library is used for parallel assignment between detections and ground truth
+        annotations.
+
+        Returns:
+            Evaluation metrics of shape (C + 1, K) where C + 1 is the number of classes.
+            plus a row for their means. K refers to the number of evaluation metrics.
+        """
+        dt_fpaths = list(self.detection_fpath.glob("*/per_sweep_annotations_amodal/*.json"))
+        gt_fpaths = list(self.ground_truth_fpath.glob("*/per_sweep_annotations_amodal/*.json"))
+
+        assert len(dt_fpaths) == len(gt_fpaths)
+        data: DefaultDict[str, np.ndarray] = defaultdict(list)
+        cls_to_ninst: DefaultDict[str, int] = defaultdict(int)
+
+        with Pool(os.cpu_count()) as p:
+            accum = p.map(self.accumulate, gt_fpaths)
+
+        for frame_stats, frame_cls_to_inst in accum:
+            for cls_name, cls_stats in frame_stats.items():
+                data[cls_name].append(cls_stats)
+            for cls_name, num_inst in frame_cls_to_inst.items():
+                cls_to_ninst[cls_name] += num_inst
+
+        data = defaultdict(np.ndarray, {k: np.vstack(v) for k, v in data.items()})
+
+        init_data = {dt_cls: MEASURE_DEFAULT_VALUES for dt_cls in self.detection_cfg.detection_classes}
+        summary = pd.DataFrame.from_dict(init_data, orient="index", columns=STATISTIC_NAMES)
+        summary_update = pd.DataFrame.from_dict(
+            self.summarize(data, cls_to_ninst), orient="index", columns=STATISTIC_NAMES
+        )
+
+        summary.update(summary_update)
+        summary = summary.round(SIGNIFICANT_DIGITS)
+        summary.index = summary.index.str.title()
+
+        summary.loc["Average Metrics"] = summary.mean().round(SIGNIFICANT_DIGITS)
+        return summary
+
+    def accumulate(self, gt_fpath: Path) -> Tuple[DefaultDict[str, np.ndarray], DefaultDict[str, int]]:
```

Comment:
also as discussed with Ben -- prefer for this to be a function, not a method, and go in utils, so we can write a simple unit test for it.